### PR TITLE
Add service worker support

### PR DESCRIPTION
**Pull request by mwistrand**
Resolves most of #96 (adding the `apple-touch-icon` meta tag requires additional voodoo that may be better suited for `webpack-contrib`). Adds service worker support when using `dist` and `dev` modes. Relying on `@dojo/webpack-contrib`'s [service worker plugin](https://github.com/dojo/webpack-contrib/#service-worker-plugin), application developers can generate a fully-featured service worker via the `.dojorc` that supports pre-caching and custom route handling (and is automatically initialized when the app starts), or they can provide the path of an existing service worker file that will be copied to the output directory.

Tested locally on both Windows and macOS with a sandbox application, and verified on Windows with `dojo eject`.

### CODE REVIEW

File: src/sw-entry.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+if ('serviceWorker' in navigator) {
```
Comment (matt-gadd) [inline src/sw-entry.ts]: Is this file needed now?

Comment (mwistrand) [reply src/sw-entry.ts]: It is not; it's in `@dojo/webpack-contrib` now. Good catch!
